sdk/python/kfp/components/_dsl_bridge.py

```diff
@@ -13,11 +13,30 @@
 # limitations under the License.
 
 from collections import OrderedDict
+from typing import Any, List, Mapping, NamedTuple
 from ._structures import ConcatPlaceholder, IfPlaceholder, InputValuePlaceholder, InputPathPlaceholder, IsPresentPlaceholder, OutputPathPlaceholder, TaskSpec
-from ._components import _generate_output_file_name, _default_component_name
+from ._components import _generate_input_file_name, _generate_output_file_name, _default_component_name
 
 
-def create_container_op_from_task(task_spec: TaskSpec):
+ResolvedContainerTask = NamedTuple(
+    'ResolvedContainerTask', [
+        ('component_name', str),                        #Name of the component
+        ('container_image', str),                       #Docker container image name
+        ('command', List[str]),                         #Command-line
+        ('arguments', str),                             #Command-line arguments
+        ('input_paths', Mapping[str, str]),             #Mapping between input names and local paths where the data will be put for use by the containerized program
+        ('input_path_arguments', Mapping[str, Any]),    #Whatever was passed as argument to the artifact inputs of the component
+        ('output_paths', Mapping[str, str]),            #Mapping between input names and local paths where the data will be put for use by the containerized program
+        ('task', TaskSpec),                             #Original TaskSpec instance
+    ]
+)
+
+
+#Holds the stack of transformation functions. The last function is called each time ResolvedContainerTask instance is created from TaskSpec.
+_created_resolved_container_task_transformation_handler = []
```

Comment:
Why do you need a stack of the handler? Are you expecting there will be use cases that user will have nested contexts?

Reply:
The contexts might be nested when there are nested pipelines (e.g. Conditionals, Loops etc)